monai/transforms/spatial/array.py

```diff
@@ -2582,7 +2593,7 @@ def __call__(
             # Flatten the first two dimensions
             strided_image = strided_image.reshape(-1, *strided_image.shape[2:])
             # Make a list of contiguous patches
-            patches = [np.ascontiguousarray(p) for p in strided_image]
+            patches = [np.copy(p) for p in strided_image]
```

Comment:
If the array is already contiguous `ascontiguousarray` should do nothing and be more efficient, otherwise it'll make a copy. Copying always makes sure the patches are detached from `strided_image`, but since this is getting reshaped into a new array anyhow this isn't an issue so I'd revert this change.

Reply:
Thanks @Nic-Ma and @ericspod for your comments. Actually there is more context that need to taken into account for this change. Here we are reshaping the array using `as_strided` and with `writeable=False` (which is recommended to avoid accidental write operations), this will cause the numpy array even when converted to torch.Tensor not be writable, so subsequent transforms on this Tensor may have undefined behavior. `ascontiguousarray` won't change writability if the array is contiguous. So, I should either make sure that we always get a writable copy of the array, or I can set `writeable=True`. The latter is more efficient but can be problematic if there is an undiscovered bug. What do you think?

Reply:
Before this line you have `strided_image = strided_image.reshape(-1, *strided_image.shape[2:])` which I think will make `strided_image` a fresh array which is writable even if the shape doesn't change. Is that correct?

Reply:
Ok so yes you're right the `copy` is needed.

Reply:
Thanks for the discussion, much clearer now.